font/src/ft/mod.rs

```diff
@@ -529,6 +581,84 @@ impl FreeTypeRasterizer {
     }
 }
 
+fn downsample_bitmap(mut bitmap_glyph: RasterizedGlyph, fixup_factor: f64) -> RasterizedGlyph {
+    // Don't try to upscale
+    if fixup_factor >= 1.0 {
+        return bitmap_glyph;
+    }
+
+    let bitmap_buffer = if let BitmapBuffer::RGBA(bitmap_buffer) = bitmap_glyph.buf {
+        bitmap_buffer
+    } else {
+        return bitmap_glyph;
+    };
```

Comment:
I think the following code is pretty easy to understand and has a little less branching you need to think about:

```rust
    // Only scale colored buffers which are bigger than required
    match (bitmap_glyph.buf, fixup_factor.cmp(&1.0)) {
        (BitmapBuffer::RGBA(buffer), Ordering::Less) => buffer,
        _ => return bitmap_glyph,
    }
```